auth-jwt-service/src/main/java/io/stargate/auth/jwt/AuthJwtService.java

```diff
@@ -0,0 +1,72 @@
+package io.stargate.auth.jwt;
+
+import com.nimbusds.jwt.JWTClaimsSet;
+import io.stargate.auth.AuthenticationService;
+import io.stargate.auth.StoredCredentials;
+import io.stargate.auth.UnauthorizedException;
+import java.text.ParseException;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AuthJwtService implements AuthenticationService {
+
+  private static final Logger logger = LoggerFactory.getLogger(AuthJwtService.class);
+
+  private final JwtValidator jwtValidator;
+  private static final String ROLE_FIELD = "x-stargate-role";
+  private static final String CLAIMS_FIELD = "stargate_claims";
+
+  public AuthJwtService(JwtValidator jwtValidator) {
+    this.jwtValidator = jwtValidator;
+  }
+
+  @Override
+  public String createToken(String key, String secret) {
+    throw new UnsupportedOperationException(
+        "Creating a token is not supported for AuthJwtService. Tokens must be created out of band.");
+  }
+
+  @Override
+  public String createToken(String key) {
+    throw new UnsupportedOperationException(
+        "Creating a token is not supported for AuthJwtService. Tokens must be created out of band.");
+  }
+
+  /**
+   * Validates a token in the form of a JWT to ensure that 1) it's not expired, 2) it's correctly
+   * signed by the provider, and 3) contains the proper role for the given DB.
+   *
+   * @param token A JWT created by an auth provider.
+   * @return A {@link StoredCredentials} containing the role name the request is authenticated to
+   *     use.
+   * @throws UnauthorizedException An UnauthorizedException if the JWT is expired, malformed, or not
+   *     properly signed.
+   */
+  @Override
+  public StoredCredentials validateToken(String token) throws UnauthorizedException {
+    JWTClaimsSet claimsSet = jwtValidator.validate(token);
+    String roleName;
+    try {
+      roleName = getRoleForJWT(claimsSet.getJSONObjectClaim(CLAIMS_FIELD));
+    } catch (ParseException e) {
+      logger.error("Failed to parse claim from JWT", e);
+      throw new RuntimeException(e);
```

Comment:
same comment about throwing raw `RuntimeException`.

Reply:
A malformed token is unauthorized. Throwing a  `UnauthorizedException` with a cause should be okay.

Reply:
Yeah I went back and forth on whether this should bubble up as a 401 or a 500. I guess "you sent me a bad JWT" makes more sense than "oh, my fault, I couldn't parse"